### ISSUE TITLE
Java: Java Sort #65

### ISSUE THREAD

**Issue by Rittika1234** [Riddhi9570/HackerrankPracticeProblems#65]
Write your solution to this java problem of hackerrank :- 
problem link : https://www.hackerrank.com/challenges/java-sort/problem

**Comment by Rittika1234** [comment]
Can you please assign me this issue for hacktoberfest?

**Comment by Riddhi9570** [comment]
@Rittika1234 Sorry, I cannot assign you multiple issues at a time. Please start working on the issue assigned to you. Thanks. All the best!

**Comment by Rittika1234** [comment]
I have completed previous issue please assign this issue to me.

**Comment by Riddhi9570** [comment]
@Rittika1234 Please do let me know how many issues you have completed till now from this repo?

**Comment by Rittika1234** [comment]
yes i have completed #64 java Substring and #20 Java Stdin and Stdout I

**Comment by adityjoshi** [comment]
hey @Riddhi9570 can I contribute to this issue

**Comment by Riddhi9570** [comment]
@adityjoshi 
Nice to know that you are eager for the contribution😃!
Unfortunately, more than one issues cannot be assigned at a time.
Please complete the previous issue.
Happy Coding😃!

**Comment by adityjoshi** [comment]
hey @Riddhi9570 kindly check the pull request of previous issue and kindly assign this issue to me

**Comment by Csrivastava003** [comment]
I want to work on this issue under hacktoberfest. @Riddhi9570

**Comment by adityjoshi** [comment]
hey @Riddhi9570 kindly assign this issue to me please

**Comment by Riddhi9570** [comment]
@adityjoshi 
it's great that you contributed to the repo twice😃🥳.
Unfortunately, I cannot assign you more than two issues for this repo for hacktoberfest. 
You may look into some other repos for hackrtoberfest contribution.
Thanks😃!All the best!

**Comment by adityjoshi** [comment]
hey @Riddhi9570 I know you have some rules but please allow me to contribute to only 2 more issues. After a lot of hard work I am able to find a repo in which I can contribute . I will feel obliged if you will allow me .

**Comment by Riddhi9570** [comment]
> hey @Riddhi9570 I know you have some rules but please allow me to contribute to only 2 more issues. After a lot of hard work I am able to find a repo in which I can contribute . I will feel obliged if you will allow me .

@adityjoshi In that case, I may help you out. Please connect with me in discord. I'm available as RB9570#5791

**Comment by yogesh-9999** [comment]
!assign

**Comment by Csrivastava003** [comment]
@Riddhi9570 I've already worked on two issues under hacktoberfest in this repository, assign it to someone else interested.

**Comment by adityjoshi** [comment]
@Riddhi9570 sent you friend request noobstar#8797 my discord id

**Comment by Riddhi9570** [comment]
> @Riddhi9570 I've already worked on two issues under hacktoberfest in this repository, assign it to someone else interested.

Thanks for letting me know! All the best for hacktoberfest 👍 !

**Comment by Riddhi9570** [comment]
@yogesh-9999 I've assigned you the issue and you may please go ahead!
Happy Coding😃!

**Comment by Mspurohit15** [comment]
Hey @Riddhi9570  assign me this issued i'm also  trying to contribute and finally on your repo

**Comment by SwaggyRajput** [comment]
Hello mam can u assign me this issue. I would love to grab this opportunity.

**Comment by 20a31a05g6** [comment]
class FactorialExample2{  
 static int factorial(int n){    
  if (n == 0)    
    return 1;    
  else    
    return(n * factorial(n-1));    
 }    
 public static void main(String args[]){  
  int i,fact=1;  
  int number=4;//It is the number to calculate factorial    
  fact = factorial(number);   
  System.out.println("Factorial of "+number+" is: "+fact);    
 }  
}





Correct the mistake in this code

Help by correcting it and resending it

Do help

Tell the mistakes in this code also

**Comment by 20a31a05g6** [comment]
> I want to work on this issue under hacktoberfest. @Riddhi9570 
> 

class FactorialExample2{  
 static int factorial(int n){    
  if (n == 0)    
    return 1;    
  else    
    return(n * factorial(n-1));    
 }    
 public static void main(String args[]){  
  int i,fact=1;  
  int number=4;//It is the number to calculate factorial    
  fact = factorial(number);   
  System.out.println("Factorial of "+number+" is: "+fact);    
 }  
}





Correct the mistake in this code

Help by correcting it and resending it

Do help

Tell the mistakes in this code also

**Comment by 20a31a05g6** [comment]
> Write your solution to this java problem of hackerrank :- 
> problem link : https://www.hackerrank.com/challenges/java-sort/problem

Please help explain line by line


class FactorialExample2{  
 static int factorial(int n){    
  if (n == 0)    
    return 1;    
  else    
    return(n * factorial(n-1));    
 }    
 public static void main(String args[]){  
  int i,fact=1;  
  int number=6;//It is the number to calculate factorial    
  fact = factorial(number);   
  System.out.println("Factorial of "+number+" is: "+fact);    
 }  
}
